example/spotLights.js

```diff
@@ -0,0 +1,753 @@
+
+			// objects
+			gltf.scene.scale.setScalar( 0.01 );
+			gltf.scene.position.x = 0;
+			gltf.scene.updateMatrixWorld();
+			gltf.castShadow = true;
+			gltf.receiveShadow = true;
+			group.add( gltf.scene );
+			gltfScenes.push( gltf.scene );
+
+			const box = new THREE.Box3();
+			box.setFromObject( gltf.scene );
+
+			gltf.scene = gltf.scene.clone();
+			gltf.scene.position.x += 8;
+			gltf.scene.updateMatrixWorld();
+			group.add( gltf.scene );
+			gltfScenes.push( gltf.scene );
+
+			gltf.scene = gltf.scene.clone();
+			gltf.scene.position.x += 8;
+			gltf.scene.updateMatrixWorld();
+			group.add( gltf.scene );
+			gltfScenes.push( gltf.scene );
+
+			gltf.scene = gltf.scene.clone();
+			gltf.scene.position.x += 8;
+			gltf.scene.updateMatrixWorld();
+			group.add( gltf.scene );
+			gltfScenes.push( gltf.scene );
+
+			gltf.scene = gltf.scene.clone();
+			gltf.scene.position.x += 8;
+			gltf.scene.updateMatrixWorld();
+			group.add( gltf.scene );
+			gltfScenes.push( gltf.scene );
+
+			const floor = new THREE.Mesh(
+				new THREE.CylinderBufferGeometry( 100, 100, 0.05, 200 ),
+				new THREE.MeshStandardMaterial( { color: 0xffffff, roughness: 0.5, metalness: 0.2 } ),
+			);
+			floor.geometry = floor.geometry.toNonIndexed();
+			floor.geometry.clearGroups();
+			floor.position.y = box.min.y - 0.05;
+			floor.receiveShadow = true;
+			group.add( floor );
+
+			const wall = new THREE.Mesh(
+				new THREE.BoxGeometry( 100, 10, 1 ),
+				new THREE.MeshStandardMaterial( { color: 0xffffff, roughness: 0, metalness: 1.0 } ),
+			);
+			wall.castShadow = true;
+			wall.receiveShadow = true;
+			wall.geometry = wall.geometry.toNonIndexed();
+			wall.geometry.clearGroups();
+			wall.position.x = 0.0;
+			wall.position.y = box.min.y + 0.05 + 5;
+			wall.position.z = box.min.z - 0.5;
+			group.add( wall );
+
+			// transform controls
+			let draggingSpotLight = false;
+
+			function makeTransformControls( object ) {
+
+				const transformControls = new TransformControls( perspectiveCamera, renderer.domElement );
+				transformControls.addEventListener( 'change', () => {
+
+					if ( draggingSpotLight ) {
+
+						ptRenderer.reset();
+
+					}
+
+				} );
+
+				transformControls.addEventListener( 'dragging-changed', ( e ) => {
+
+					draggingSpotLight = e.value;
+					return controls.enabled = ! e.value;
+
+				} );
+
+				transformControls.attach( object );
+				transformControlsScene.add( transformControls );
+
+			}
+
+			// spot lights
+			spotLightHelpers = [ ];
+			lights = [ ];
+
+			const decays = [ 0, 1.5, 0, 0.25, 0 ];
+			for ( let i = 0; i < 5; ++ i ) {
+
+				const spotLight = new THREE.SpotLight( 0xffffff );
+
+				spotLight.position.set( i * 8, 7.0, 0.005 );
+				spotLight.angle = Math.PI / 8.0;
+				spotLight.penumbra = 0.0;
+				spotLight.decay = decays[ i ];
+				spotLight.distance = 0.0;
+				spotLight.intensity = 50.0;
+				spotLight.lampIntensityScale = 0.0;
+				spotLight.radius = 0.5;
+				spotLight.iesProfile = - 1 + i;
```

Comment:
Similar to the `PhysicalCamera` implementation that includes DoF parameters I'm wondering if it makes sense to include a `PhysicalSpotlight` class that includes IES and radius parameters.

Reply:
That sounds good.